rolecall/rolecall.py

```diff
@@ -146,14 +145,14 @@ def _get_emoji_from_entry(self, entry: Entry):
         server = ctx.message.server
         author = ctx.message.author
 
-        # check if role was mentioned 
+        # check if role was mentioned
+        role_object = await self.get_or_create("role", role, server) 
         nums_found = re.findall('\d+', role)
         if nums_found:
             potential_role_id = nums_found[0]
             if potential_role_id in [r.id for r in server.roles]:
+                await self.bot.delete_role(server, role_object)
                 role_object = discord.utils.get(server.roles, id=potential_role_id)
-        else:
-            role_object = await self.get_or_create("role", role, server)
```

Comment:
whhhhhat?
you're getting or creating a role, but then if it already exists, you're deleting the role and then trying to get it again?

Reply:
No. Check if the passed argument contains a role id(AKA a role mention) and if it does, delete the role that was created earlier and just get the role associated with the id found